packages/sdk/app-framework/src/plugins/SurfacePlugin/Surface.tsx

```diff
@@ -84,6 +87,20 @@ export type SurfaceProps = PropsWithChildren<{
 export const Surface = forwardRef<HTMLElement, SurfaceProps>(
   ({ role, name = role, fallback, placeholder, ...rest }, forwardedRef) => {
     const props = { role, name, fallback, ...rest };
+    const { debugInfo } = useSurfaceRoot();
+
+    // Track debug info.
+    const [id] = useState<string>(Math.random().toString(36).slice(2));
+    useEffect(() => {
+      debugInfo?.set(id, { id, created: Date.now(), name, role, renderCount: 0 });
+      return () => {
+        debugInfo?.delete(id);
+      };
+    }, [id]);
+    if (debugInfo?.get(id)) {
+      debugInfo.get(id)!.renderCount++;
+    }
```

Comment:
do you have a sense of how much overhead this adds when it's not being used?

Reply:
would react's `useId` work instead of creating a custom id?

Reply:
i think it should be unmeasurable. there's deliberately no state change or refresh.